### ISSUE TITLE
Support for fetch in nodejs

### ISSUE THREAD

**Issue by CEbbinghaus** [rustwasm/wasm-bindgen#3377]
### Motivation

[fetch](https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch) has been part of the nodejs runtime since v17.5.0 (behind an experimental flag) and v18.0.0 (without the flag) and as such is available in the "global" scope in both web and node. Furthermore, it is also available in the global scope in both deno and bun.js making it almost entirely cross runtime/environment. As such `wasm-bindgen` should also support fetch without referencing the `window` object as is shown in the [docs](https://rustwasm.github.io/wasm-bindgen/examples/fetch.html)

### Proposed Solution

allowing access to the fetch method via the "global" scope in any runtime, This would be the window object for browser & the `global` object for nodejs. (not too sure about deno & bun.js)

### Alternatives

Leave it as is and limit using fetch to only the browser

### Additional Context

[Old Outdated forum post saying it doesn't exist in node](https://users.rust-lang.org/t/wasm-bindgen-fetch-example-doesnt-work-on-nodejs/48417)
[New forum post created by me in parallel to this issue](https://users.rust-lang.org/t/adding-support-for-global-fetch-in-wasm/91830)

**Comment by daxpedda** [comment]
`wasm-bindgen` wouldn't fit the bill, this would probably belong into `web-sys`, but it's specifically made for Web APIs, not Node.js.

So the missing part here is probably something like `node-sys`, which would then be able to implement the Node.js global object and all it's methods. This can be a completely standalone library and doesn't need to live in this repository.

Considering the developer and maintenance bandwidth available, this is very much out of scope.

**Comment by stan-irl** [comment]
@CEbbinghaus just bind straight fetch as an external. if its there the call will be successful. Here is an example: https://github.com/devashishdxt/tonic-web-wasm-client/blob/main/src/fetch.rs

```
use web_sys::{Request, RequestInit, Response};
use js_sys::Promise;
use wasm_bindgen::prelude::wasm_bindgen;
#[wasm_bindgen]
extern "C" {
    #[wasm_bindgen(js_name = fetch)]
    fn fetch_with_request_and_init(input: &Request, init: &RequestInit) -> Promise;
}
```
